src/Common/src/Interop/Unix/System.Native/Interop.Poll.cs

```diff
@@ -20,9 +21,10 @@ internal enum PollEvents : short
 
         internal struct PollEvent
         {
-            internal int FileDescriptor;         // The file descriptor to poll
+            internal IntPtr FileDescriptor;      // The file descriptor to poll
```

Comment:
I personally do not think that this is necessary. As you've pointed out, its only benefit is consistency and it does come at a not insignificant cost.